smartapps/smartthings/ifttt.src/ifttt.groovy

```diff
@@ -131,19 +131,68 @@ def update() {
 	def type = params.deviceType
 	def data = request.JSON
 	def devices = settings[type]
+	def device = settings[type].find { it.id == params.id }
 	def command = data.command
 
-	log.debug "[PROD] update, params: ${params}, request: ${data}, devices: ${devices*.id}"
-	if (command) {
-		def device = devices?.find { it.id == params.id }
+    log.debug "[PROD] update, params: ${params}, request: ${data}, devices: ${devices*.id}"
+	if (validateCommand(device, type, command)) {
 		if (!device) {
 			httpError(404, "Device not found")
 		} else {
 			device."$command"()
 		}
+	} else {
+		httpError(403, "Access denied. This command is not supported by current capability.")
+    }
+}
+
+/**
+ * Validating the command passed by the user based on capability.
+ * @return boolean
+ */
+def validateCommand(device, deviceType, command) {
+	def capabilityCommands = getDeviceCapabilityCommands(device.capabilities)
```

Comment:
If device is null, there will be an exception here and code will never reach the 404 "Device not found"